solr/core/src/test/org/apache/solr/cloud/TestTlogReplica.java

```diff
@@ -142,9 +145,8 @@ private void assertUlogPresence(DocCollection collection) {
         try {
           core = cluster.getReplicaJetty(r).getCoreContainer().getCore(r.getCoreName());
           assertNotNull(core);
-          assertTrue(
-              "Update log should exist for replicas of type Append",
-              new java.io.File(core.getUlogDir()).exists());
+          File tlogDir = getHypotheticalTlogDir(core);
```

Comment:
is there test with `UpdateLog` 's dir as well ? Just want to cover regular and ulog-dir test

Reply:
You're right to point this out; I think tests are inadequate in this area (hence the longstanding regression!). Definitely want to build this out before merging -- definitely before potentially merging upstream.